src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && herbivore.hidden === false) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(herbivore), 1);
```

Comment:
[CODE PERFOMANCE] - The use of `splice` in combination with `indexOf` can be inefficient when removing an animal from the `Animal.alive` array. It's better to use `filter` to create a new array excluding the dead animal. This avoids nested loops and can improve performance.